### DISCUSSION TITLE
Automatically switch profile when .exe is launched.

### DISCUSSION THREAD

**Discussion by sw0277** [T-Troll/alienfx-tools#329] [Q&A] [answered]
I created a new profile then set it up to automatically switch to it when I launch a game. Everything was working as intended and would automatically switch to the new profile. But, after my pc was shut down and sat over night, its no longer switching automatically when I launch a game. However, I can switch it manually.

The "Profile auto switch" setting in the main settings is enabled. 
The "Priority profile" for the new profile is enabled.
The "Fan settings" for the new profile is enabled.
I added multiple .exe files both games and random programs to the "Applications" box for the new profile.

I tried enabling "Only then active" for the new profile as well. I uninstalled then reinstalled alienfx to completely start from scratch and still can't get it to automatically switch. 

Alienfx-tools version 8.2.6.2
Windows 11 Pro 22H2 currently up to date.
Dell G5SE 5505 Ryzen 7 4800H / RX 5600M 

EDIT: I'm also noticing that when the keyboard RGB times out, the RGB wont come back on when pressing a key or using the touch pad. I have to manually switch between the 2 profiles before the RGB will come on again.. This wasn't an issue until I created a second profile. If I only have the default profile, the RGB will work as expected and come back on after it times out.

Any suggestions?

**Comment by T-Troll** [comment, marked as answer]
Hmm... Interesting.
What do you mean "shut down"? Sleep? Hibernate? Full power off?

Be careful with "Priority"/"Active only"  - you can make the system stuck at one of the profiles forever (f.e. BOTH profiles are priority = no switch).

Also, keep in mind - auto switch disabled if application window visible on screen. It only activates, then app minimized.

About timeout - need more data as well. What do you mean? There is no timeout in app, just "follow screen state" - in this mode, app intercept screen/lid state change and switch according to these changes. Profiles can't interfere with it (except dimming), it's a separate switch.

**Comment by sw0277** [comment]
When I said “shut down” I meant full power off.

If I minimize the application, the  profile does switch automatically so it is working correctly. That's really weird because it was switching profiles with the application visible on the screen when I first set it up and was testing things. I never would have thought you had to minimize it in order for it to work. Either way I know this now and it works fine.

What I meant by timeout is, if I walk away from my pc for 15 min the RGB on my keyboard times out, it shuts off. Under normal circumstances if you push a key or touch the touchpad the RGB comes back on. Now, once the RGB goes off it does not come back on by pressing a key or touching the touchpad. I have to manually switch profiles back and forth then the RGB comes back on.  

And when you say "Be careful with "Priority"/"Active only" - you can make the system stuck at one of the profiles forever.", do you mean this will be permanent at the device/hardware level? As in it can't be reversed?

**Reply by T-Troll** [reply]
Got it, so it's working correctly. The reason is profile should not change then you edit it, so i block change if app not minimized.

So it's another timeout (i hear some device have it). Hmm... I can switch brightness back to set, but by what event (definitely not keypress)? So think about how to define this situation and then you need to leave it.

No, it's not hardware. Just the algorithm i use for profile selection. "Priority" (if app running) can be only overrided by "Priority"+"Active" and app foreground. Other way it will not be changed.

**Comment by sw0277** [comment]
Ok, now I understand.. I was able to set the RGB on my keyboard to "never" timeout when connected to AC power in the bios. So that should solve my issue with that for now. Everything's working good now, thanks.

**Reply by T-Troll** [reply]
I'll also add brightness refresh to "Refresh" button/menu at next release for these situations.

**Comment by sw0277** [comment]
The newest version 8.2.6.7 has fixed the issue with the RGB not coming back on after timing out.. It works well now..

**Reply by T-Troll** [reply]
Glad to hear it.

**Comment by sw0277** [comment]
The newest version 8.3.0 has caused the RGB keyboard light issue that I previously had, that version 8.2.6.7 had fixed. After the RGB keyboard lights time out (RGB lights on keyboard turn off after 15 min of inactivity), will not turn back on by keypress or using the touchpad. You have to use the "refresh" button or completely close Alienfx-tools before the RGB will turn back on..
